pkg/xds/context/mesh_context_builder.go

```diff
@@ -583,3 +574,11 @@ func inferServiceProtocol(endpoints []xds.Endpoint) core_mesh.Protocol {
 	}
 	return serviceProtocol
 }
+
+func isDestinationResource(resources core_model.ResourceList) bool {
```

Comment:
This is exactly stuff that should be in the type descriptor. I think there should be a `IsDestination` akin to `IsProxy`: https://github.com/kumahq/kuma/blob/e279bba108307837bbe966f032c5e95039759372/pkg/core/resources/model/resource.go#L204

If you are worried about type explosion turning these 2 in flags like we did for KDS may make things simpler.

Advantage of this is that you could also ensure at generation that the types implement the `Destination` interface by generating a `var _ Destination = MyType{}`. Thus helping anyone that adds a destination in the future